src/geometry/euclidean/circumcenter.lean

```diff
@@ -377,12 +377,64 @@ begin
           (λ i, hr i (set.mem_univ _))).symm
 end
 
+/-- The orthogonal projection of a point `p` onto the hyperplane spanned by the simplex's points. -/
+def orthogonal_projection_span {n : ℕ} (s : simplex ℝ P n) :=
+orthogonal_projection (affine_span ℝ (set.range s.points))
+
+/-- Adding a vector to a point in the given subspace, then taking the
+orthogonal projection, produces the original point if the vector is a
+multiple of the result of subtracting a point's orthogonal projection
+from that point. -/
+lemma orthogonal_projection_vadd_smul_vsub_orthogonal_projection
+  {n : ℕ} (s : simplex ℝ P n) {p1 : P} (p2 : P) (r : ℝ)
+  (hp : p1 ∈ affine_span ℝ (set.range s.points)) :
+  s.orthogonal_projection_span (r • (p2 -ᵥ s.orthogonal_projection_span p2 : V) +ᵥ p1) =
+    ⟨p1, hp⟩ :=
+orthogonal_projection_vadd_smul_vsub_orthogonal_projection _ _ _
+
+lemma orthogonal_projection_span_eq {n : ℕ} {r₁ : ℝ} (s : simplex ℝ P n)
+  {p p₁ : P} {p₁o : P} (hp₁o : p₁o ∈ affine_span ℝ (set.range s.points))
+  (hp₁ : p₁ = r₁ • (p -ᵥ ↑(s.orthogonal_projection_span p)) +ᵥ p₁o) :
+  ↑(s.orthogonal_projection_span p₁) = p₁o :=
+begin
+  have := s.orthogonal_projection_vadd_smul_vsub_orthogonal_projection _ _ hp₁o,
+  rw ← hp₁ at this,
+  convert congr_arg coe this
+end
```

Comment:
This is kind of a weird lemma, as presumably you could start with `subst hp₁` at the call site - it's a bit like having `(hx : x = 0) : x + y = x`.